blobstorage/blobstorage.go

```diff
@@ -0,0 +1,58 @@
+package blobstorage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"google.golang.org/grpc/metadata"
+)
+
+func UploadFile(ctx context.Context, uploadURL string, data []byte, contentType string) error {
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, uploadURL, nil)
+
+	if err != nil {
+		return fmt.Errorf("creating request: %w", err)
+	}
+
+	body := bytes.NewReader(data)
+	contentLength := int64(len(data))
+	req.Body = io.NopCloser(body)
+	req.Header = metadataToHTTPHeaders(ctx)
+
+	req.ContentLength = contentLength
+	req.Header.Set("Content-Type", contentType)
+	req.Header.Del("Authorization")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	if err != nil {
+		return fmt.Errorf("uploading blob: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("upload failed with status %d: %s", resp.StatusCode, string(body))
```

Comment:
nit: I'd add the body (even the status code) as a separate log. You want to keep errors that come from the same path as consistent as possible. That allows you, e.g., to group errors to see how frequent an incident is.

Reply:
I am not sure if I got it correctly. 
Do you mean like [this](https://github.com/instill-ai/x/pull/27/commits/48880040d83296052e5fd3a91cc54aa94877b837#diff-f08f67e3874ac803e2c4620a878a9f7d79bfc0ab9e1e6b7f91b11493c8abd833R41)?

Reply:
Sort of, I'd align with the logger package we use in other repos:

```go
	err := fmt.Errorf("failed to upload file")
	logger.Error("Failed to upload file to MinIO",
		zap.String("body", string(body)),
		zap.Int("status", resp.StatusCode),
		zap.Error(err),
	)
	return err
```

Reply:
[updated commit](https://github.com/instill-ai/x/pull/27/commits/342f816ae8c35cef2531f729bf66ddeb2e81c3c7).
[update pipeline-backend](https://github.com/instill-ai/pipeline-backend/pull/844).